src/spacy/app.clj

```diff
@@ -28,18 +29,55 @@
               :else x))]
     (walk/postwalk walk event)))
 
+(def events
+  {"Strategie Event Open Space 2020" (bidi/path-for routes ::event :event-slug "dezember-2020-strategie-event")})
+
+(html/deftemplate index-template "templates/index.html"
+  []
+  [:ul [:li html/first-of-type]] (html/clone-for [[caption url] events]
+                                                 [:li :a] (html/content caption)
+                                                 [:li :a] (html/set-attr :href url)))
+
 (defn index [system]
   (handler-util/get-resource
    (fn [ctx]
-     (selmer/render-file
-      "templates/index.html"
-      {:links [{:href (bidi/path-for routes ::event :event-slug "dezember-2020-strategie-event")
-                :text "Strategie Event Open Space 2020"}]}))))
+     (apply str (index-template)))))
+
+(defn current-status [is-next-up next-up]
+  (cond
+    is-next-up ::up-next
+    next-up    ::please-wait
+    :else      ::nobody-in-queue))
+
+(defn status-map [title]
+  {::up-next [:span "You are currently next in line! Please " [:a {:href "#sessions"} "select a slot"] " for your session \"" title "\""]
+   ::please-wait "Please wait for others to present their session"
+   ::nobody-in-queue "There are currently no sessions in the queue"})
+
+(html/defsnippet up-next "templates/event/up-next.html"
+  [:up-next]
+  [{:keys [current-user is-next-up next-up
+           session-title statuses status]
+    :or   {session-title  (get-in next-up [:spacy.domain/session :spacy.domain/title])
+           statuses       (status-map session-title)
+           status         (current-status is-next-up next-up)}}]
```

Comment:
@jstepien enlive seems to work pretty heavily with macros. I wasn't sure how to best create variables with a snippet scope, so I'm using `:or` here to do something that I would normally use a `let` for:

```clojure
(let [session-title (get-in next-up [:spacy.domain/session :spacy.domain/title])
      statuses (status-map session-title)
      status (current-status is-next-up next-up)]
  ;; do something with the variables here
  )
```

Do you know a better way to do this?

Reply:
If I recall correctly, there's no easier way to introduce variables with a snippet scope.

You could consider introducing a function with that `let` you'd normally use. That function would build a map with all necessary data and pass it to the snippet. Something to the effect of:

```clojure
(defn up-next [{:keys [current-user is-next-up next-up ,,,]}
  (let [session-title (get-in next-up [:spacy.domain/session :spacy.domain/title])
        statuses (status-map session-title)
        status (current-status is-next-up next-up)
        values {,,,} ;; a map with all the snippet needs
        ]
    (up-next-snippet values))

(html/defsnippet up-next-snippet "templates/event/up-next.html"
  ,,,)
```

This leads to a simpler snippet. It gets all the data it needs and it's responsible only for correct templating.

Reply:
Ah... thank you. That does seem more readable. I've modified this for the `up-next` snippet